tests/framework/session/test_session_extension_hooks.py

```diff
@@ -562,6 +578,34 @@ def test_after_dataset_load_hook_async(
             ["Before dataset loaded", "Catalog load", "After dataset loaded"]
         ).strip("[]") in str(hooks_log_messages).strip("[]")
 
+    def test_after_dataset_load_hook_async_multiple_outputs(
+        self,
+        mocker,
+        memory_catalog,
+        mock_session,
+        sample_node_multiple_outputs,
+    ):
+        # load mock context to instantiate Hooks
+        mock_session.load_context()
```

Comment:
Could we minimise the dependency tree of the test here by just mocking the `hook_manger` and not mock the whole session? `_run_node_async` doesn't need anything else from the session.

Reply:
Good point, I hadn't thought of this. Fixed in [08c0c9e](https://github.com/kedro-org/kedro/pull/2159/commits/08c0c9e7d50373961ca4be9e9c620a1406c991ee).

The existing test (`test_after_dataset_load_hook_async`) still requires a `mock_session` and can't be modified in the same way.